src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,377 @@
+
+package com.autotune.analyzer.recommendations.updater.vpa;
+
+import com.autotune.analyzer.exceptions.ApplyRecommendationsError;
+import com.autotune.analyzer.exceptions.UnableToCreateVPAException;
+import com.autotune.analyzer.kruizeObject.KruizeObject;
+import com.autotune.analyzer.recommendations.RecommendationConfigItem;
+import com.autotune.analyzer.recommendations.updater.RecommendationUpdaterImpl;
+import com.autotune.analyzer.recommendations.utils.RecommendationUtils;
+import com.autotune.analyzer.utils.AnalyzerConstants;
+import com.autotune.analyzer.utils.AnalyzerErrorConstants;
+import com.autotune.common.k8sObjects.K8sObject;
+import com.autotune.common.data.result.ContainerData;
+import com.autotune.analyzer.recommendations.objects.MappedRecommendationForTimestamp;
+import com.autotune.analyzer.recommendations.objects.TermRecommendations;
+import io.fabric8.autoscaling.api.model.v1.*;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionList;
+import io.fabric8.kubernetes.api.model.autoscaling.v1.CrossVersionObjectReferenceBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.ApiextensionsAPIGroupDSL;
+import io.fabric8.verticalpodautoscaler.client.DefaultVerticalPodAutoscalerClient;
+import io.fabric8.verticalpodautoscaler.client.NamespacedVerticalPodAutoscalerClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VpaUpdaterImpl extends RecommendationUpdaterImpl {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VpaUpdaterImpl.class);
+    private static VpaUpdaterImpl vpaUpdater;
+
+    private KubernetesClient kubernetesClient;
+    private ApiextensionsAPIGroupDSL apiextensionsClient;
+
+
+    private VpaUpdaterImpl() {
+        this.kubernetesClient = new DefaultKubernetesClient();
+        this.apiextensionsClient = kubernetesClient.apiextensions();
+    }
+
+    public static VpaUpdaterImpl getInstance() {
+        if (null != vpaUpdater) {
+            return vpaUpdater;
+        }
+
+        synchronized (VpaUpdaterImpl.class) {
+            if (null == vpaUpdater) {
+                vpaUpdater = new VpaUpdaterImpl();
+            }
+        }
+
+        return vpaUpdater;
+    }
+
+    /**
+     * Checks whether the necessary updater dependencies are installed or available in the system.
+     * @return boolean true if the required updaters are installed, false otherwise.
+     */
+    @Override
+    public boolean isUpdaterInstalled() {
+        try {
+            LOGGER.info(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CHECKING_IF_UPDATER_INSTALLED,
```

Comment:
General rule to follow, only log it as `info` if it is a rare occurence, eg once per kruize start. All other items which are on a per experiment basis should only be logged as `debug` unless there is a justification

Reply:
Thanks for pointing it out @dinogun! I’ve updated the `info` logs to `debug` messages and ensured that only `error` messages are logged by default. I’ll ensure this in future implementations also.